ironfish/src/rpc/routes/wallet/createTransaction.ts

```diff
@@ -0,0 +1,187 @@
+   CreateTransactionRequestSchema,
+   async (request, node): Promise<void> => {
+     const transaction = request.data
+ 
+     const account = node.wallet.getAccountByName(transaction.fromAccountName)
+ 
+     if (!account) {
+       throw new ValidationError(`No account found with name ${transaction.fromAccountName}`)
+     }
+ 
+     // The node must be connected to the network first
+     if (!node.peerNetwork.isReady) {
+       throw new ValidationError(
+         `Your node must be connected to the Iron Fish network to send a transaction`,
+       )
+     }
+ 
+     if (!node.chain.synced) {
+       throw new ValidationError(
+         `Your node must be synced with the Iron Fish network to send a transaction. Please try again later`,
+       )
+     }
+ 
+     const receives = transaction.receives.map((receive) => {
+       let assetId = Asset.nativeId()
+       if (receive.assetId) {
+         assetId = Buffer.from(receive.assetId, 'hex')
+       }
+ 
+       return {
+         publicAddress: receive.publicAddress,
+         amount: CurrencyUtils.decode(receive.amount),
+         memo: receive.memo,
+         assetId,
+       }
+     })
+ 
+     const totalByAssetIdentifier = new BufferMap<bigint>()
+     if(transaction.fee){
+        const fee = CurrencyUtils.decode(transaction.fee)
+        if (fee < 1n) {
+          throw new ValidationError(`Invalid transaction fee, ${transaction.fee}`)
+        }
+
+        totalByAssetIdentifier.set(Asset.nativeId(), fee)
+    }
+ 
+     
+     for (const { assetId, amount } of receives) {
+       if (amount < 0) {
+         throw new ValidationError(`Invalid transaction amount ${amount}.`)
+       }
+ 
+       const sum = totalByAssetIdentifier.get(assetId) ?? BigInt(0)
+       totalByAssetIdentifier.set(assetId, sum + amount)
+     }
+ 
+     // Check that the node account is updated
+     for (const [assetId, sum] of totalByAssetIdentifier) {
+       const balance = await node.wallet.getBalance(account, assetId)
+ 
+       if (balance.confirmed < sum) {
+         throw new ValidationError(
+           `Your balance is too low. Add funds to your account first`,
+           undefined,
+           ERROR_CODES.INSUFFICIENT_BALANCE,
+         )
+       }
+     }
+
+     let rawTransaction
+ 
+     try {
+       if(transaction.fee){
+        rawTransaction = await node.wallet.createTransaction(
+            account,
+            receives,
+            [],
+            [],
+            BigInt(transaction.fee),
+            transaction.expirationDelta ?? node.config.get('transactionExpirationDelta'),
+            transaction.expiration,
+        )
+       }else{
+        let priprityLevel: PriorityLevel
+        switch (transaction.feePriorityLevel) {
+            case 'low':
+              priprityLevel = 'low'
+              break
+            case 'high':
+              priprityLevel = 'high'
+              break
+            default:
+              priprityLevel = 'medium'
```

Comment:
Is the `switch` necessary? It looks like its only effect is to default to `'medium'`

Reply:
so `transaction.feePriorityLevel` is type of string. And priprityLevel has enum type of "low" | "medium" | "high". I can not directly convert because Type 'string' is not assignable to type '"low" | "medium" | "high"'